### PR TITLE
Adds keycloak-gatekeeper as OIDC Authenticator

### PR DESCRIPTION
**Pull request by krishnadurai**
**Which issue is resolved by this Pull Request:**
Related issues: #267 

**Description of your changes:**
This PR introduces Keycloak Gatekeeper as an OIDC Authenticator for Kubeflow central dashboard.

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate`
    3. `make test`

/cc @swiftdiaries 
/assign @krishnadurai 

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/337)

### CODE REVIEW

File: dex-auth/keycloak-gatekeeper/base/virtualservice.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: keycloak-gatekeeper
+spec:
+  gateways:
+  - kubeflow/kubeflow-gateway
```
Comment (swiftdiaries) [inline dex-auth/keycloak-gatekeeper/base/virtualservice.yaml]: Is the `kubeflow/kubeflow-gateway` supposed to refer the `kubeflow-gateway` in the `kubeflow` namespace?

**Comment by swiftdiaries** [comment]
Could you please add to the [README](https://github.com/kubeflow/manifests/blob/master/dex-auth/README.md) to reference where keycloak fits in?

File: dex-auth/keycloak-gatekeeper/base/virtualservice.yaml (hunk 1 above)
Comment (krishnadurai) [reply dex-auth/keycloak-gatekeeper/base/virtualservice.yaml]: Yes this is kubeflow namespace's kubeflow gateway. The naming style is namespace/gateway if the gateway referred to is not in the same namespace.
https://github.com/istio/istio/issues/11131

**Comment by krishnadurai** [comment]
@swiftdiaries I'll add these in the README for dex-auth folder. This is an Identification Proxy like IAP for kubeflow.
Right now the way to install this would be to do a kustomize build. I'm going to introduce installation through kfctl after this PR gets in.

**Comment by krishnadurai** [comment]
@swiftdiaries I've added the README documentation as requested.

**Pull request merged**